docs/compilation.md

```diff
@@ -26,7 +26,71 @@ Next, create the `pg_duckdb` extension:
 CREATE EXTENSION pg_duckdb;
 ```
 
-TODO: List instructions to build 'from scratch', including all required dependencies with a `brew`/`apt install`, on at least:
+# Ubuntu 24.04
 
-* Ubuntu 24.04
-* MacOS
+This example uses Postgres 17. If you wish to use another version, substitute the version number in the commands as necessary.
+
+### Set up Postgres
+
+We recommend using PGDG for Postgres, but you are welcome to use any Postgres packages or install from source. To install Postgres 17 from PGDG:
+
+```sh
+sudo apt install postgresql-common
+sudo /usr/share/postgresql-common/pgdg/apt.postgresql.org.sh
+sudo apt install postgresql-17 postgresql-server-dev-17
+```
+
+If you do not install from PGDG, please note that you must have the `dev` package installed to compile extensions.
+
+### Install Build Dependencies
+
+```sh
+sudo apt install \
+    build-essential libreadline-dev zlib1g-dev flex bison libxml2-dev \
+    libxslt-dev libssl-dev libxml2-utils xsltproc pkg-config libc++-dev \
+    libc++abi-dev libglib2.0-dev libtinfo6 cmake libstdc++-12-dev \
+    liblz4-dev ninja-build
+```
+
+### Clone, Build, and Install pg_duckdb
+
+```sh
+git clone https://github.com/duckdb/pg_duckdb
+cd pg_duckdb
+```
+
+```sh
+make -j16
+sudo make install
+```
+
+### Add pg_duckdb to shared_preload_libraries
+
+```sh
+sudo -s
+echo "shared_preload_libraries = 'pg_duckdb'" >/etc/postgresql/17/main/conf.d/pg_duckdb.conf
+exit
```

Comment:
Interesting, I hadn't seen this approach before. I'm more used to the `sudo tee` approach.
```sh
echo "shared_preload_libraries = 'pg_duckdb'" | sudo tee /etc/postgresql/17/main/conf.d/pg_duckdb.conf
```

This seems fine too though.

Reply:
changed to `tee` in 7de00c3b92342622656d247fd278ac3668abb2d1, thanks!